packages/nextjs/app/_components/Grants/index.tsx

```diff
@@ -1,18 +1,33 @@
 import { ApprovedGrantsList } from "./ApprovedGrantsList";
 import { getPublicGrants } from "~~/services/database/repositories/grants";
+import { PublicGrant } from "~~/services/database/repositories/grants";
+import { getPublicLargeGrants } from "~~/services/database/repositories/large-grants";
+import { PublicLargeGrant } from "~~/services/database/repositories/large-grants";
+
+// Add a discriminator property to distinguish between PublicGrant and PublicLargeGrant
+type DiscriminatedGrant = (PublicGrant & { type: "grant" }) | (PublicLargeGrant & { type: "largeGrant" });
```

Comment:
type is duplicated, same is in the `ApprovedGrantsList`. I think it's better to move it to some common `types` file. Also I'm thinking maybe it's worth to add `type` to db so it will not needed to add it on the frontend? (don't forget to remove adding it on frontend if decide to add to schema)

Reply:
Moved to types/utils!